src/main/java/uk/gov/hmcts/ccd/domain/service/common/StateReferenceService.java

```diff
@@ -0,0 +1,50 @@
+package uk.gov.hmcts.ccd.domain.service.common;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Map;
+import java.util.Optional;
+import javax.inject.Inject;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.ccd.domain.casestate.EnablingConditionParser;
+import uk.gov.hmcts.ccd.domain.model.definition.EventPostStateDefinition;
+
+@Service
+public class StateReferenceService {
+
+    private final EnablingConditionParser enablingConditionParser;
+
+    @Inject
+    public StateReferenceService(EnablingConditionParser enablingConditionParser) {
+        this.enablingConditionParser = enablingConditionParser;
+    }
+
+    public Optional<String> evaluatePostStateCondition(List<EventPostStateDefinition> eventPostStateDefinitions,
+                                                       Map<String, JsonNode> caseEventData) {
+        List<EventPostStateDefinition> postStateDefinitions = Optional
+            .ofNullable(eventPostStateDefinitions)
+            .orElse(new ArrayList<>());
```

Comment:
don't think it's possible in CCD to not have a post state right? maybe we can get rid of this? seems unnecessary boilerplate code

Reply:
But when we have * defined in PostStateCondition, I don't think we are saving that in DB....